### PR TITLE
Update LIP 0055

### PR DESCRIPTION
**Pull request by ricott1**

### CODE REVIEW

File: proposals/lip-0055.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ The second change is to update the block header schema. In general, it is desira
 - does not need to be changed when modules are added or removed to a running blockchain,
 - is used by every blockchain in the Lisk ecosystem regardless of the modules implemented in the individual chains.
 
-Furthermore, we update the block header schema to include new properties introduced by [LIP 0040][lip-0040], [LIP 0058][lip-0058], [LIP 0061][lip-0061], and [LIP "Introduce events and event root"][research:introduce-events].
+Furthermore, we update the block header schema to include new properties introduced by [LIP 0040][lip-0040], [LIP 0058][lip-0058], [LIP 0061][lip-0061], and [LIP "Introduce events and event root"][lip-0065].
```
Comment (janhack) [inline proposals/lip-0055.md:32]:
```suggestion
Furthermore, we update the block header schema to include new properties introduced by [LIP 0040][lip-0040], [LIP 0058][lip-0058], [LIP 0061][lip-0061], and [LIP 0065][lip-0065].
```

File: proposals/lip-0055.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ This LIP introduces the following new block header properties:
 
 - `generatorAddress`: The address of the block generator. It replaces the `generatorPublicKey` property. See [below](#change-generator-public-key-to-generator-address) for more details.
 - `assetRoot`: The root of the Merkle tree computed from the block assets array. See [below](#separation-between-block-header-and-block-assets) for more details.
-- `eventRoot`: The root of the sparse Merkle tree that is computed from the events emitted during the block processing. See [LIP "Introduce events and event root"][research:introduce-events] LIP for the reason why it needs to be included in a block header.
+- `eventRoot`: The root of the sparse Merkle tree that is computed from the events emitted during the block processing. See [LIP "Introduce events and event root"][lip-0065] LIP for the reason why it needs to be included in a block header.
```
Comment (janhack) [inline proposals/lip-0055.md:44]:
```suggestion
- `eventRoot`: The root of the sparse Merkle tree that is computed from the events emitted during the block processing. See [LIP 0065][lip-0065] LIP for the reason why it needs to be included in a block header.
```

File: proposals/lip-0055.md [hunk 3]
```diff
@@ -590,7 +597,7 @@ verifyEventRoot():
   return block.header.eventRoot == eventRoot(block.header.height)
 ```
 
-Here, the function `eventRoot` calculates the events root of the block with the input height as specified in [LIP "Introduce events and event root"][research:introduce-events].
+Here, the function `eventRoot` calculates the events root of the block with the input height as specified in [LIP "Introduce events and event root"][lip-0065].
```
Comment (janhack) [inline proposals/lip-0055.md:600]:
```suggestion
Here, the function `eventRoot` calculates the events root of the block with the input height as specified in [LIP 0065][lip-0065].
```

**Review by janhack — APPROVED** [review]
Just some cosmetic suggestions, otherwise looks good.

**Review by sergeyshemyakov — APPROVED** [review]
Looks good to me

File: proposals/lip-0055.md [hunk 4]
```diff
@@ -201,13 +203,13 @@ The block is validated in the [static validation stage](#block-processing-stages
...
 ```python
-blockID():
+def blockID():
   # Check that the signature length is 64 bytes to ensure
   # that we do not compute the block ID of an unsigned block
   if length(block.header.signature) != SIGNATURE_LENGTH_EDDSA_BYTES:
     return error
 
-  let serializedBlockHeader be the serialization of block.header following the blockHeaderSchema
+  serializedBlockHeader encode(blockHeaderSchema, block.header)
```
Comment (janhack) [inline proposals/lip-0055.md:212]:
```suggestion
  serializedBlockHeader = encode(blockHeaderSchema, block.header)
```

File: proposals/lip-0055.md [hunk 5]
```diff
@@ -622,18 +629,16 @@ The aggregate commit contains an aggregate BLS signature of a certificate corres
...
   generatorKey = validators.getValidatorAccount(block.header.generatorAddress).generatorKey
   signature = block.header.signature
 
   # Remove the signature from the block header
   delete block.header.signature
   # Serialize the block header without signature
-  let serializedUnsignedBlockHeader be the serialization of block.header following the unsignedBlockHeaderSchema
-
-  let networkIdentifier be the network identifier of the chain
+  serializedUnsignedBlockHeader encode(unsignedBlockHeaderSchema, block.header)
```
Comment (janhack) [inline proposals/lip-0055.md:639]:
```suggestion
  serializedUnsignedBlockHeader = encode(unsignedBlockHeaderSchema, block.header)
```

**Review by janhack — CHANGES_REQUESTED** [review]
I left a suggestion for two small typos.

**Review by janhack — APPROVED** [review]
Looks all good.

**Review by karmacoma — APPROVED** [review]

**Pull request merged**